lifecycle/lifecycle/server/metrics.py

```diff
@@ -23,6 +23,12 @@ def setup_lifecycle_metrics():
     REGISTRY.register(DatabaseConnectionCollector())
 
 
+def unregister_metrics():
+    collectors = list(REGISTRY._collector_to_names.keys())
+    for collector in collectors:
+        REGISTRY.unregister(collector)
```

Comment:
Why is such function needed - for performance?

Reply:
This is needed for testing. This was pretty weird when debugging. In general, test cases were complaining that prometheus metrics are already registered. Apparently, pytest reuses some parts when running multiple tests in a test suite, thus tests were colliding. Actually, it revealed the bug - not cleaning up properly and now it's fixed.

Reply:
Alright. It's good idea to add such info as comment, so coder knows its purpose.